packages/frontend/src/components/DataViz/store/lineChartSlice.ts

```diff
@@ -10,14 +13,15 @@ export const lineChartConfigSlice = createSlice({
         ...cartesianChartConfigSlice.caseReducers,
     },
     extraReducers: (builder) => {
-        builder.addCase(onResults, (state, action) => {
+        builder.addCase(setChartOptionsAndConfig, (state, action) => {
             if (action.payload.type !== ChartKind.LINE) {
                 return;
             }
 
             state.options = action.payload.options;
 
-            if (!state.config) {
+            // Only set the initial config if it's not already set and the fieldConfig is present
```

Comment:
This was made mostly because previously we would set the state `columns` whenever we set the fields, but now we only do it when running the query (similar to how sql runner does it) - see `packages/frontend/src/features/semanticViewer/store/semanticViewerSlice.ts`

The initial chart configuration is only set `onResults` (now called `setChartOptionsAndConfig`) which would mean that because there were no columns then the initial config would not have a `fieldConfig` the first time it runs for the default chart type (`bar`).

Also this initial config is only set if it is `undefined`, which would mean that if we set the config without `fieldConfig` then it wouldn't be able to set it again and therefore the bar chart would always have an incorrect configuration